MDANSE_GUI/Src/MDANSE_GUI/Tabs/Plotters/Grid.py

```diff
@@ -144,6 +142,13 @@ def plot(
             except KeyError:
                 _, best_axis = dataset.longest_axis()
             for key, curve in islice(dataset._curves.items(), self._plot_limit):
+                counter += 1
+                if counter > self._plot_limit:
+                    LOG.warning(
+                        "Curves above the current limit of %s will be ignored",
+                        self._plot_limit,
+                    )
+                    break
```

Comment:
These blocks will never be hit because of the `islice`. Also `counter` should be `enumerate` if it's needed.

Reply:
Do we want to `chain` them, then `islice` then?